From 95eb8d23ce4bec610ab868a8277f2ff4511ceb24 Mon Sep 17 00:00:00 2001
From: Peter Marschall <peter@adpm.de>
Date: Wed, 1 Jun 2011 13:41:25 +0200
Subject: [PATCH 03/26] OpenPGP: immediately quit on allocation errors in init


Signed-off-by: Peter Marschall <peter@adpm.de>
---
 src/libopensc/card-openpgp.c |   13 ++++---------
 1 files changed, 4 insertions(+), 9 deletions(-)

diff --git a/src/libopensc/card-openpgp.c b/src/libopensc/card-openpgp.c
index 4eaefb0..9c792df 100644
--- a/src/libopensc/card-openpgp.c
+++ b/src/libopensc/card-openpgp.c
@@ -324,18 +324,13 @@ pgp_init(sc_card_t *card)
 		    (info->get_fn != NULL)) {
 			child = pgp_new_blob(priv->mf, info->id, info->type, info);
 			/* catch out of memory condition */
-			if (child == NULL)
-				break;
+			if (child == NULL) {
+				pgp_finish(card);
+				return SC_ERROR_OUT_OF_MEMORY;
+			}
 		}
 	}
 
-	/* treat out of memory condition */
-	if (child == NULL) {
-		pgp_finish(card);
-		return SC_ERROR_OUT_OF_MEMORY;
-	}
-
-
 	/* get card_features from ATR & DOs */
 	pgp_get_card_features(card);
 
-- 
1.7.5.3

