From 8a0cb4550b222a17959b1338dbcd172565632aaa Mon Sep 17 00:00:00 2001
From: Peter Marschall <peter@adpm.de>
Date: Sun, 13 Mar 2011 19:04:29 +0100
Subject: [PATCH 03/32] OpenPGP: try to match flags with specification

---
 src/libopensc/card-openpgp.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/libopensc/card-openpgp.c b/src/libopensc/card-openpgp.c
index 11e51c4..b3f3d62 100644
--- a/src/libopensc/card-openpgp.c
+++ b/src/libopensc/card-openpgp.c
@@ -150,7 +150,9 @@ pgp_init(sc_card_t *card)
 	card->cla = 0x00;
 
 	/* Is this correct? */
+	/* OpenPGP card spec 1.1 & 2.0, section 2.1 */
         flags = SC_ALGORITHM_RSA_RAW;
+	/* OpenPGP card spec 1.1 & 2.0, section 7.2.9 & 7.2.10 */
         flags |= SC_ALGORITHM_RSA_PAD_PKCS1;
         flags |= SC_ALGORITHM_RSA_HASH_NONE;
 
-- 
1.7.4.4

