From 8700041b3a976beebe5243008fb3e0bd9dbd3a2d Mon Sep 17 00:00:00 2001
From: Peter Marschall <peter@adpm.de>
Date: Wed, 1 Jun 2011 21:07:13 +0200
Subject: [PATCH 10/26] opensc-tool: no unnecessary spaces around "DF"


Signed-off-by: Peter Marschall <peter@adpm.de>
---
 src/tools/opensc-tool.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/tools/opensc-tool.c b/src/tools/opensc-tool.c
index 075380b..f55120e 100644
--- a/src/tools/opensc-tool.c
+++ b/src/tools/opensc-tool.c
@@ -332,13 +332,13 @@ static int print_file(sc_card_t *in_card, const sc_file_t *file,
 		tmps = "iEF";
 		break;
 	case SC_FILE_TYPE_DF:
-		tmps = " DF";
+		tmps = "DF";
 		break;
 	default:
 		tmps = "unknown";
 		break;
 	}
-	printf("type: %-3s, ", tmps);
+	printf("type: %s, ", tmps);
 	if (file->type != SC_FILE_TYPE_DF) {
 		const id2str_t ef_type_name[] = {
 			{ SC_FILE_EF_TRANSPARENT,         "transparent"           },
-- 
1.7.5.3

